### PR TITLE
Revert "build: Make install rules optionnal"

### PR DESCRIPTION
**Pull request by rzr**
This will simplify packaging recipes, it's not mandatory
but there is no reason to keep this,
if is does not imply any regressions.

This reverts commit 272148dbdc40aac9c9446a3a9bcc6511f5eb8bbf.
Relate-to: https://github.com/lvgl/lvgl/issues/2534

### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Checkpoints
- [ ] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Update [CHANGELOG.md](https://github.com/lvgl/lvgl/blob/master/docs/CHANGELOG.md)
- [ ] Update the documentation

### CODE REVIEW

**Review by embeddedt — APPROVED** [review]
As long as lines 117-120 don't cause any strange behavior I'm fine with removing it.

**Review by embeddedt — CHANGES_REQUESTED** [review]
It seems that some of the changes in #2621 may be breaking ESP32 Micropython as @amirgon feared: https://github.com/lvgl/lvgl/pull/2629#issuecomment-933865415

Please do not merge for now until we figure out the exact issue.

**Comment by rzr** [comment]
is there any CI log to look at ?


https://github.com/lvgl/lvgl/pull/2629/checks?check_run_id=3796564110

 /home/runner/work/lvgl/lvgl/lib/lv_bindings/lvgl/CMakeLists.txt:1
  (project):

    project command is not scriptable

Ok i can change this it's not related to the install feature  sorry for that

**Comment by rzr** [comment]
will rebase on https://github.com/lvgl/lvgl/pull/2629 once merged

**Comment by rzr** [comment]
Actually it's harmless to merge now, It will simplify my bitbake recipes:

https://github.com/lvgl/lvgl/pull/2636

cc: @embeddedt

**Comment by rzr** [comment]
I just rebased to check CI it should be ok to merge now

cc: @embeddedt

**Comment by embeddedt** [comment]
@rzr Thanks, could I trouble you to do one last rebase, to bring https://github.com/lvgl/lvgl/commit/a625dc220621cd512f1dd41554968481a6bc9e65 in?

**Comment by rzr** [comment]
No luck:

 CMake Error at /home/runner/work/lvgl/lvgl/lib/lv_bindings/lvgl/CMakeLists.txt:114 (set_target_properties):
  set_target_properties Can not find target to add properties to: lvgl

OK I realize now that no static lib is declared for all platforms (including ESP) , I wonder why?

**Comment by embeddedt** [comment]
Isn't it declared [here](https://github.com/lvgl/lvgl/blob/ea2e9d7f4656030be8eb565083e914e7b63efe9f/CMakeLists.txt#L88-L89), and as an interface library in the other case as well?

Only ESP32 and Zephyr skip it, from what I see, and I think neither of those platforms have their own component logic and don't link with a CMake static library directly.

Maybe it'd be cleanest to just move the install rules under [this `else` block](https://github.com/lvgl/lvgl/blob/ea2e9d7f4656030be8eb565083e914e7b63efe9f/CMakeLists.txt#L69).

**Comment by rzr** [comment]
yea I figured this out, let's relocate it then

**Comment by embeddedt** [comment]
Looks good to me; let's see if CI passes on it.

**Comment by embeddedt** [comment]
Thanks, I've merged this, hopefully it simplifies the Bitbake portion of things.

**Comment by rzr** [comment]
A bit thx for your reactivity

**Pull request merged**
